<style>
    @keyframes textAnimation10 {
        0% { color: #FFA500; } /* Orange */
        50% { color: #FF6347; } /* Tomato */
        100% { color: #FF7F50; } /* Coral */
    }
</style>

<div style="border: 4px solid #FFA500; border-radius: 45px; padding: 45px; box-shadow: 0 9px 18px rgba(0, 0, 0, 0.7); background: linear-gradient(45deg, #FFFFFF, #FFDAB9); max-width: 750px; margin: 50px auto;">
    <p style="font-family: 'Garamond', serif; font-size: 55px; text-align: center; color: #FFA500; font-style: italic; font-weight: bold; animation: textAnimation10 9s infinite alternate;">Video Game Sales Exploratory Data Analysis</p>
</div>


![](https://media1.tenor.com/m/nqbZejon51sAAAAC/video-games-tom-ellis.gif)

<div style='background-color: #f0f0f0; color: #333; border: 2px solid #ccc; border-radius: 5px; padding: 15px;'>

# <span style='color:#E1B12D;'>Meta Data</span><br>

<i>This dataset comprises a collection of video games with sales exceeding 100,000 copies. It was obtained through scraping <a href="https://www.vgchartz.com/"><b>vgchartz.com</b></a>.</i><br>

<b>Fields include</b><br>

   -  <b>Name</b> - The game's title<br>
   -  <b>Platform</b> - The platform of the game's release (e.g., PC, PS4)<br>
   - <b>Year</b> - The year of the game's release<br>
   - <b>Genre</b> - The genre of the game<br>
   - <b>Publisher</b> - The game's publisher<br>
   - <b>NA_Sales</b> - Sales in North America (in millions)<br>
   - <b>EU_Sales</b> - Sales in Europe (in millions)<br>
   - <b>JP_Sales</b> - Sales in Japan (in millions)<br>
   - <b>Other_Sales</b> - Sales in the rest of the world (in millions)<br>
   - <b>Global_Sales</b> - Total worldwide sales<br>

<i>The script used to scrape the data is available at <a href="https://github.com/GregorUT/vgchartzScrape">GitHub</a>. It is based on BeautifulSoup using Python. There are 16,598 records. Two records were dropped due to incomplete information.</i><br>

# <span style='color:#E1B12D;'>Objective</span><br>

<i>Data analysis can be categorized into two fields: Hypothesis Generation and Hypothesis Confirmation (also called confirmatory


----

<div style="border: 3px solid orange; border-radius: 20px; padding: 10px; width: fit-content;">
    <div style="background: linear-gradient(45deg, #FFFFFF, #FFDAB9); border-radius: 20px; padding: 30px; font-family: Arial, sans-serif; font-size: 28px; color: #FFA500; display: inline-block; text-align: center; box-shadow: 0 5px 15px rgba(0, 0, 0, 0.3);">
        <span style="font-weight: bold; margin-right: 10px;">1</span>
        <span style="color: #FF6347; margin-right: 10px; font-size: 42px;">🌠</span>
        <span style="font-style: roboto; font-weight: bold;">Import Packages</span>
    </div>
</div>


In [2]:
import pandas as pd 
import numpy as np 
import seaborn as sns
import matplotlib.pyplot as plt 
from tabulate import tabulate

<div style="border: 3px solid orange; border-radius: 20px; padding: 10px; width: fit-content;">
    <div style="background: linear-gradient(45deg, #FFFFFF, #FFDAB9); border-radius: 20px; padding: 30px; font-family: Arial, sans-serif; font-size: 28px; color: #FFA500; display: inline-block; text-align: center; box-shadow: 0 5px 15px rgba(0, 0, 0, 0.3);">
        <span style="font-weight: bold; margin-right: 10px;">2</span>
        <span style="color: #FF6347; margin-right: 10px; font-size: 42px;">🌠</span>
        <span style="font-style: roboto; font-weight: bold;">Import Dataset</span>
    </div>
</div>


In [3]:
df = pd.read_csv('/kaggle/input/global-video-game-sales/vgsales.csv')

In [4]:
def display_df_info(df):
    # Display shape
    rows, cols = df.shape
    print(f"╔{'═'*30}╗")
    print(f"║{'DataFrame Shape':^30}║")
    print(f"╠{'═'*30}╣")
    print(f"║{'Rows:':<10}{rows:>10}          ║")
    print(f"║{'Columns:':<10}{cols:>10}          ║")
    print(f"╚{'═'*30}╝")


    # Display data types
    print(f"\n╔{'═'*30}╗")
    print(f"║{'DataFrame Column Data Types':^30}║")
    print(f"╠{'═'*30}╣")
    for col, dtype in df.dtypes.items():
        print(f"║{col:<20}{str(dtype):>10}║")
    print(f"╚{'═'*30}╝")

# Example usage
display_df_info(df)


╔══════════════════════════════╗
║       DataFrame Shape        ║
╠══════════════════════════════╣
║Rows:          16598          ║
║Columns:          11          ║
╚══════════════════════════════╝

╔══════════════════════════════╗
║ DataFrame Column Data Types  ║
╠══════════════════════════════╣
║Rank                     int64║
║Name                    object║
║Platform                object║
║Year                   float64║
║Genre                   object║
║Publisher               object║
║NA_Sales               float64║
║EU_Sales               float64║
║JP_Sales               float64║
║Other_Sales            float64║
║Global_Sales           float64║
╚══════════════════════════════╝


In [5]:
df.head(10)


,Rank,Name,Platform,Year,Genre,Publisher,NA_Sales,EU_Sales,JP_Sales,Other_Sales,Global_Sales
0,1,Wii Sports,Wii,2006.0,Sports,Nintendo,41.49,29.02,3.77,8.46,82.74
1,2,Super Mario Bros.,NES,1985.0,Platform,Nintendo,29.08,3.58,6.81,0.77,40.24
2,3,Mario Kart Wii,Wii,2008.0,Racing,Nintendo,15.85,12.88,3.79,3.31,35.82
3,4,Wii Sports Resort,Wii,2009.0,Sports,Nintendo,15.75,11.01,3.28,2.96,33.00
4,5,Pokemon Red/Pokemon Blue,GB,1996.0,Role-Playing,Nintendo,11.27,8.89,10.22,1.00,31.37
5,6,Tetris,GB,1989.0,Puzzle,Nintendo,23.20,2.26,4.22,0.58,30.26
6,7,New Super Mario Bros.,DS,2006.0,Platform,Nintendo,11.38,9.23,6.50,2.90,30.01
7,8,Wii Play,Wii,2006.0,Misc,Nintendo,14.03,9.20,2.93,2.85,29.02
8,9,New Super Mario Bros. Wii,Wii,2009.0,Platform,Nintendo,14.59,7.06,4.70,2.26,28.62
9,10,Duck Hunt,NES,1984.0,Shooter,Nintendo,26.93,0.63,0.28,0.47,28.31


---
<div style="margin-top: 20px; font-family: Arial, sans-serif; font-size: 22px; color: #FFA500; font-weight: bold;">2.1 || Dealing with Missing values</div>

---


In [6]:
df.isnull().sum()

Rank              0
Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64


> <div style="color: orange; font-weight: bold; font-size: larger;">We got Missing Values In Year and Publisher Column</div>

<div style="color: orange; margin-top: 10px;"><b>How I will deal with these Missing Values</b></div>

<div style="color: orange; margin-top: 10px;">Publisher</div>
<ul>
    <li><span style="color: orange;">Strategy:</span> I simply replace NaN (Null Values) with a string ('Unknown') instead of removing that column because I am doing EDA and it will provide some information.</li>
</ul>

<div style="color: orange; margin-top: 10px;">Year</div>
<ul>
    <li><span style="color: orange;">Strategy:</span> I simply Use Digit 0000 to deal with Null values because there are 271 null values. In this way, I will not lose actual information and provide insights from the real dataset without changing anything out of it.</li>
</ul>


In [7]:
# Or use the alternative suggested by the warning
df['Publisher'] = df['Publisher'].fillna('Unknown')


In [8]:
# df['Year'].fillna(0000, inplace=True)
df['Year'] = df['Year'].fillna(0)


In [12]:
# it is compulsory to change the data type because we got some unusual values like this ===> 2002.0 2023.0 to ===> 2002 , 2023 
df['Year'] = df['Year'].astype(int)

In [13]:
df.isnull().sum()

Rank            0
Name            0
Platform        0
Year            0
Genre           0
Publisher       0
NA_Sales        0
EU_Sales        0
JP_Sales        0
Other_Sales     0
Global_Sales    0
dtype: int64


> <div style="color: orange; font-weight: bold; font-size: larger;">Successfully deal with Missing Values </div>


---
<div style="margin-top: 20px; font-family: Arial, sans-serif; font-size: 22px; color: #FFA500; font-weight: bold;">2.2 ||  Checking and dealing With Wrong Data Type </div>

---



In [11]:
df.head(10)

,Rank,Name,Platform,Year,Genre,Publisher,NA_Sales,EU_Sales,JP_Sales,Other_Sales,Global_Sales
0,1,Wii Sports,Wii,2006,Sports,Nintendo,41.49,29.02,3.77,8.46,82.74
1,2,Super Mario Bros.,NES,1985,Platform,Nintendo,29.08,3.58,6.81,0.77,40.24
2,3,Mario Kart Wii,Wii,2008,Racing,Nintendo,15.85,12.88,3.79,3.31,35.82
3,4,Wii Sports Resort,Wii,2009,Sports,Nintendo,15.75,11.01,3.28,2.96,33.00
4,5,Pokemon Red/Pokemon Blue,GB,1996,Role-Playing,Nintendo,11.27,8.89,10.22,1.00,31.37
5,6,Tetris,GB,1989,Puzzle,Nintendo,23.20,2.26,4.22,0.58,30.26
6,7,New Super Mario Bros.,DS,2006,Platform,Nintendo,11.38,9.23,6.50,2.90,30.01
7,8,Wii Play,Wii,2006,Misc,Nintendo,14.03,9.20,2.93,2.85,29.02
8,9,New Super Mario Bros. Wii,Wii,2009,Platform,Nintendo,14.59,7.06,4.70,2.26,28.62
9,10,Duck Hunt,NES,1984,Shooter,Nintendo,26.93,0.63,0.28,0.47,28.31


In [12]:
df.dtypes

Rank              int64
Name             object
Platform         object
Year              int64
Genre            object
Publisher        object
NA_Sales        float64
EU_Sales        float64
JP_Sales        float64
Other_Sales     float64
Global_Sales    float64
dtype: object


---
<div style="margin-top: 20px; font-family: Arial, sans-serif; font-size: 22px; color: #FFA500; font-weight: bold;">2.3 ||  Dealing With Column [Name]</div>

---



In [13]:
df['Name'].unique

<bound method Series.unique of 0                                              Wii Sports
1                                       Super Mario Bros.
2                                          Mario Kart Wii
3                                       Wii Sports Resort
4                                Pokemon Red/Pokemon Blue
                               ...                       
16593                  Woody Woodpecker in Crazy Castle 5
16594                       Men in Black II: Alien Escape
16595    SCORE International Baja 1000: The Official Game
16596                                          Know How 2
16597                                    Spirits & Spells
Name: Name, Length: 16598, dtype: object>

In [14]:
#  There is a number of games having their version like Taken 3 , Taken 7 so if i remove digits from it then i will lost information so i have tried this simple but effective way to deal with this 
#  problem in this way i will not lost any information plus point is i would be able to keep it in correct datatype :)  
df['Name'] = df['Name'].str.replace('5', 'Five')
df['Name'] = df['Name'].str.replace('9', 'Nine')
df['Name'] = df['Name'].str.replace('8', 'Eight')
df['Name'] = df['Name'].str.replace('3', 'Three')
df['Name'] = df['Name'].str.replace('2', 'Two')
df['Name'] = df['Name'].str.replace('7', 'Seven')
df['Name'] = df['Name'].str.replace('6', 'Six')
df['Name'] = df['Name'].str.replace('4', 'Four')
df['Name'] = df['Name'].str.replace('1', 'One')
df['Name'] = df['Name'].str.replace('0', 'Zero')


In [15]:
# Example: Remove non-alphanumeric characters from the 'Name' column 
df['Name'] = df['Name'].replace('[^a-zA-Z0-9\s]', '', regex=True)


In [16]:
# Example: Remove numeric characters from the 'Name' column
df['Name'] = df['Name'].replace('\d', '', regex=True)


In [17]:
# Example: Display unique values in the 'Name' column
df['Name']


0                                               Wii Sports
1                                         Super Mario Bros
2                                           Mario Kart Wii
3                                        Wii Sports Resort
4                                  Pokemon RedPokemon Blue
                               ...                        
16593                Woody Woodpecker in Crazy Castle Five
16594                         Men in Black II Alien Escape
16595    SCORE International Baja OneZeroZeroZero The O...
16596                                         Know How Two
16597                                      Spirits  Spells
Name: Name, Length: 16598, dtype: object

In [18]:
df['Name'] = df['Name'].str.strip()


In [19]:
df.dtypes

Rank              int64
Name             object
Platform         object
Year              int64
Genre            object
Publisher        object
NA_Sales        float64
EU_Sales        float64
JP_Sales        float64
Other_Sales     float64
Global_Sales    float64
dtype: object

In [20]:
#  printing all unique values to manually check the data from a human eye with eagle eyesight abilities 
unique_names = df['Name'].unique()

# Display unique values using a loop
for name in unique_names:
    print(name)


Wii Sports

Super Mario Bros

Mario Kart Wii

Wii Sports Resort

Pokemon RedPokemon Blue

Tetris

New Super Mario Bros

Wii Play

New Super Mario Bros Wii

Duck Hunt

Nintendogs

Mario Kart DS

Pokemon GoldPokemon Silver

Wii Fit

Wii Fit Plus

Kinect Adventures

Grand Theft Auto V

Grand Theft Auto San Andreas

Super Mario World

Brain Age Train Your Brain in Minutes a Day

Pokemon DiamondPokemon Pearl

Super Mario Land

Super Mario Bros Three

Grand Theft Auto Vice City

Pokemon RubyPokemon Sapphire

Pokemon BlackPokemon White

Brain Age Two More Training in Minutes a Day

Gran Turismo Three ASpec

Call of Duty Modern Warfare Three

Pokmon Yellow Special Pikachu Edition

Call of Duty Black Ops

Pokemon XPokemon Y

Call of Duty Black Ops Three

Call of Duty Black Ops II

Call of Duty Modern Warfare Two

Grand Theft Auto III

Super Smash Bros Brawl

Animal Crossing Wild World

Mario Kart Seven

Halo Three

Pokemon HeartGoldPokemon SoulSilver

Super Mario SixFour

Gran Turismo Four

Sup

In [21]:
unique_names.shape

(11470,)

In [22]:
df['Name'].duplicated().sum()

5128

In [23]:
df.drop_duplicates(subset=['Name'], inplace=True)


In [24]:
df['Name'].duplicated().sum()

0



---
<div style="margin-top: 20px; font-family: Arial, sans-serif; font-size: 22px; color: #FFA500; font-weight: bold;">2.4 || Dealing With Column [Platform]</div>

---



In [25]:
df['Platform']

0        Wii
1        NES
2        Wii
3        Wii
4         GB
        ... 
16589    PSV
16590    PSP
16592     DS
16593    GBA
16596     DS
Name: Platform, Length: 11470, dtype: object

In [26]:
df['Platform'] = df['Platform'].str.replace('3', 'Three')
df['Platform'] = df['Platform'].str.replace('2', 'Tow')
df['Platform'] = df['Platform'].str.replace('6', 'Six')
df['Platform'] = df['Platform'].str.replace('4', 'Four')
df['Platform'] = df['Platform'].str.replace('1', 'One')
df['Platform'] = df['Platform'].str.replace('0', 'Zero')

In [27]:
df['Platform'] = df['Platform'].replace('[^a-zA-Z0-9\s]', '', regex=True)


In [28]:
def display_value_counts(df, column_name):
    value_counts = df[column_name].value_counts()
    print(f"╔{'═'*30}╗")
    print(f"║{'Value Counts':^30}║")
    print(f"╠{'═'*30}╣")
    for index, count in value_counts.items():
        print(f"║{str(index):<20}{str(count):>10}║")
    print(f"╚{'═'*30}╝")

# Example usage
display_value_counts(df, 'Platform')


╔══════════════════════════════╗

║         Value Counts         ║

╠══════════════════════════════╣

║PSTow                     1855║

║DS                        1805║

║PS                        1121║

║Wii                        948║

║PSP                        904║

║PSThree                    705║

║GBA                        619║

║XThreeSixZero              559║

║PC                         495║

║ThreeDS                    396║

║PSV                        280║

║XB                         276║

║NSixFour                   245║

║PSFour                     241║

║SNES                       228║

║GC                         170║

║SAT                        148║

║TowSixZeroZero             125║

║NES                         89║

║GB                          84║

║WiiU                        55║

║DC                          43║

║XOne                        30║

║GEN                         23║

║NG                           9║

║SCD                          6║

║WS           

In [29]:
df['Platform'] = df['Platform'].astype('str')


### ***Dealing With Column [Genre]***

In [30]:
display_value_counts(df, 'Genre')


╔══════════════════════════════╗

║         Value Counts         ║

╠══════════════════════════════╣

║Action                    1917║

║Sports                    1375║

║Misc                      1327║

║Role-Playing              1218║

║Adventure                 1047║

║Shooter                    815║

║Racing                     766║

║Simulation                 725║

║Fighting                   612║

║Platform                   588║

║Strategy                   582║

║Puzzle                     498║

╚══════════════════════════════╝


In [31]:
df['Genre'] = df['Genre'].str.replace('-', ' ')


In [32]:
df['Genre'].unique()

array(['Sports', 'Platform', 'Racing', 'Role Playing', 'Puzzle', 'Misc',
       'Shooter', 'Simulation', 'Action', 'Fighting', 'Adventure',
       'Strategy'], dtype=object)


---
<div style="margin-top: 20px; font-family: Arial, sans-serif; font-size: 22px; color: #FFA500; font-weight: bold;">2.5 || Dealing With Column [Publisher]</div>

---



In [33]:
df['Publisher'].unique()

array(['Nintendo', 'Microsoft Game Studios', 'Take-Two Interactive',
       'Sony Computer Entertainment', 'Activision', 'Ubisoft',
       'Bethesda Softworks', 'Electronic Arts', 'Sega', 'SquareSoft',
       'Atari', '505 Games', 'Capcom', 'GT Interactive',
       'Konami Digital Entertainment',
       'Sony Computer Entertainment Europe', 'Square Enix', 'LucasArts',
       'Virgin Interactive', 'Warner Bros. Interactive Entertainment',
       'Universal Interactive', 'Eidos Interactive', 'RedOctane',
       'Vivendi Games', 'Enix Corporation', 'Namco Bandai Games',
       'Palcom', 'Hasbro Interactive', 'THQ', 'Fox Interactive',
       'Acclaim Entertainment', 'MTV Games', 'Disney Interactive Studios',
       'Unknown', 'Majesco Entertainment', 'Codemasters', 'Red Orb',
       'Level 5', 'Midway Games', 'JVC', 'Deep Silver', '989 Studios',
       'NCSoft', 'UEP Systems', 'Maxis', 'Imagic', 'Tecmo Koei',
       'Valve Software', 'ASCII Entertainment', 'Mindscape', 'Infogrames',
      

In [34]:
df['Publisher'] = df['Publisher'].str.replace('5', 'Five')
df['Publisher'] = df['Publisher'].str.replace('9', 'Nine')
df['Publisher'] = df['Publisher'].str.replace('8', 'Eight')
df['Publisher'] = df['Publisher'].str.replace('3', 'Three')
df['Publisher'] = df['Publisher'].str.replace('2', 'Tow')
df['Publisher'] = df['Publisher'].str.replace('7', 'Seven')
df['Publisher'] = df['Publisher'].str.replace('6', 'Six')
df['Publisher'] = df['Publisher'].str.replace('4', 'Four')
df['Publisher'] = df['Publisher'].str.replace('1', 'One')
df['Publisher'] = df['Publisher'].str.replace('0', 'Zero')

In [35]:
df['Publisher'] = df['Publisher'].replace('[^a-zA-Z0-9\s]', '', regex=True)


In [36]:
df['Publisher'].unique()

array(['Nintendo', 'Microsoft Game Studios', 'TakeTwo Interactive',
       'Sony Computer Entertainment', 'Activision', 'Ubisoft',
       'Bethesda Softworks', 'Electronic Arts', 'Sega', 'SquareSoft',
       'Atari', 'FiveZeroFive Games', 'Capcom', 'GT Interactive',
       'Konami Digital Entertainment',
       'Sony Computer Entertainment Europe', 'Square Enix', 'LucasArts',
       'Virgin Interactive', 'Warner Bros Interactive Entertainment',
       'Universal Interactive', 'Eidos Interactive', 'RedOctane',
       'Vivendi Games', 'Enix Corporation', 'Namco Bandai Games',
       'Palcom', 'Hasbro Interactive', 'THQ', 'Fox Interactive',
       'Acclaim Entertainment', 'MTV Games', 'Disney Interactive Studios',
       'Unknown', 'Majesco Entertainment', 'Codemasters', 'Red Orb',
       'Level Five', 'Midway Games', 'JVC', 'Deep Silver',
       'NineEightNine Studios', 'NCSoft', 'UEP Systems', 'Maxis',
       'Imagic', 'Tecmo Koei', 'Valve Software', 'ASCII Entertainment',
       'Minds


> <div style="margin-top: 20px; font-family: Arial, sans-serif; font-size: 28px; color: #FFA500; font-weight: bold;"> Congrats !! We have Cleaned All Categorical Data</div>







---
<div style="margin-top: 20px; font-family: Arial, sans-serif; font-size: 22px; color: #FFA500; font-weight: bold;">2.6 || Dealing With Numerical Columns</div>

---



In [37]:
df.info()

<class 'pandas.core.frame.DataFrame'>

Index: 11470 entries, 0 to 16596

Data columns (total 11 columns):

 #   Column        Non-Null Count  Dtype  

---  ------        --------------  -----  

 0   Rank          11470 non-null  int64  

 1   Name          11470 non-null  object 

 2   Platform      11470 non-null  object 

 3   Year          11470 non-null  int64  

 4   Genre         11470 non-null  object 

 5   Publisher     11470 non-null  object 

 6   NA_Sales      11470 non-null  float64

 7   EU_Sales      11470 non-null  float64

 8   JP_Sales      11470 non-null  float64

 9   Other_Sales   11470 non-null  float64

 10  Global_Sales  11470 non-null  float64

dtypes: float64(5), int64(2), object(4)

memory usage: 1.1+ MB


In [38]:
df.head()

,Rank,Name,Platform,Year,Genre,Publisher,NA_Sales,EU_Sales,JP_Sales,Other_Sales,Global_Sales
0,1,Wii Sports,Wii,2006,Sports,Nintendo,41.49,29.02,3.77,8.46,82.74
1,2,Super Mario Bros,NES,1985,Platform,Nintendo,29.08,3.58,6.81,0.77,40.24
2,3,Mario Kart Wii,Wii,2008,Racing,Nintendo,15.85,12.88,3.79,3.31,35.82
3,4,Wii Sports Resort,Wii,2009,Sports,Nintendo,15.75,11.01,3.28,2.96,33.00
4,5,Pokemon RedPokemon Blue,GB,1996,Role Playing,Nintendo,11.27,8.89,10.22,1.00,31.37


<div style="border: 3px solid orange; border-radius: 20px; padding: 10px; width: fit-content;">
    <div style="background: linear-gradient(45deg, #FFFFFF, #FFDAB9); border-radius: 20px; padding: 30px; font-family: Arial, sans-serif; font-size: 28px; color: #FFA500; display: inline-block; text-align: center; box-shadow: 0 5px 15px rgba(0, 0, 0, 0.3);">
        <span style="font-weight: bold; margin-right: 10px;">3</span>
        <span style="color: #FF6347; margin-right: 10px; font-size: 42px;">🌠</span>
        <span style="font-style: roboto; font-weight: bold;">Deep Dive
    </div>
</div>


> <div style="margin-top: 12px; font-family: Arial, sans-serif; font-size: 17px; color: #FFA500; font-weight: bold;">Sales are paramount in every market, serving as a crucial metric for assessing performance and guiding strategic decisions. Therefore, our forthcoming analysis will focus primarily on the sales data of the top 15 performers over the past five years.</div>


***Total fractional sales***:
- Well the sales are in miilions so lets take a look :) 

In [39]:
df['Year'].max()

2020

In [40]:
!pip install tabulate





---
<div style="margin-top: 20px; font-family: Arial, sans-serif; font-size: 22px; color: #FFA500; font-weight: bold;"> 3.1 || Top 10 Worldwide Sales Games</div>

---



In [41]:
# Filter the DataFrame for years between 2015 and 2020
filtered_df = df[(df['Year'] >= 2015) & (df['Year'] <= 2020)]

# Group by 'Year', 'Name', 'Platform', 'Genre', and 'Publisher', and sum the 'Global_Sales'
top_10_games_global_sales = filtered_df.groupby(['Year', 'Name', 'Platform', 'Genre', 'Publisher'])['Global_Sales'].sum().nlargest(15)
top_10_games_global_sales



Year  Name                                   Platform  Genre         Publisher                            
2015  Call of Duty Black Ops Three           PSFour    Shooter       Activision                               14.24
      FIFA OneSix                            PSFour    Sports        Electronic Arts                           8.49
      Star Wars Battlefront TwoZeroOneFive   PSFour    Shooter       Electronic Arts                           7.67
      Fallout Four                           PSFour    Role Playing  Bethesda Softworks                        6.96
2016  FIFA OneSeven                          PSFour    Sports        Electronic Arts                           4.77
2015  Splatoon                               WiiU      Shooter       Nintendo                                  4.57
      Uncharted The Nathan Drake Collection  PSFour    Action        Sony Computer Entertainment               4.47
      Halo Five Guardians                    XOne      Shooter       Microsoft Ga

|    |   Year | Name                                   | Platform   | Genre        | Publisher                             |   Global_Sales |
|----|--------|----------------------------------------|------------|--------------|---------------------------------------|----------------|
|  0 |   2015 | Call of Duty Black Ops Three           | PSFour     | Shooter      | Activision                            |          14.24 |
|  1 |   2015 | FIFA OneSix                            | PSFour     | Sports       | Electronic Arts                       |           8.49 |
|  2 |   2015 | Star Wars Battlefront TwoZeroOneFive   | PSFour     | Shooter      | Electronic Arts                       |           7.67 |
|  3 |   2015 | Fallout Four                           | PSFour     | Role Playing | Bethesda Softworks                    |           6.96 |
|  4 |   2016 | FIFA OneSeven                          | PSFour     | Sports       | Electronic Arts                       |           4.77 |
|  5 |   2015 | Splatoon                               | WiiU       | Shooter      | Nintendo                              |           4.57 |
|  6 |   2015 | Uncharted The Nathan Drake Collection  | PSFour     | Action       | Sony Computer Entertainment           |           4.47 |
|  7 |   2015 | Halo Five Guardians                    | XOne       | Shooter      | Microsoft Game Studios                |           4.26 |
|  8 |   2016 | Uncharted Four A Thiefs End            | PSFour     | Shooter      | Sony Computer Entertainment           |           4.20 |
|  9 |   2015 | NBA TwoKOneSix                         | PSFour     | Sports       | TakeTwo Interactive                   |           3.85 |
| 10 |   2015 | Batman Arkham Knight                   | PSFour     | Action       | Warner Bros Interactive Entertainment |           3.79 |
| 11 |   2015 | The Witcher Three Wild Hunt            | PSFour     | Role Playing | Namco Bandai Games                    |           3.73 |
| 12 |   2016 | Tom Clancys The Division               | PSFour     | Shooter      | Ubisoft                               |           3.61 |
| 13 |   2015 | Metal Gear Solid V The Phantom Pain    | PSFour     | Action       | Konami Digital Entertainment          |           3.38 |
| 14 |   2015 | Assassins Creed Syndicate              | PSFour     | Action       | Ubisoft                               |           3.28 |





---
<div style="margin-top: 20px; font-family: Arial, sans-serif; font-size: 22px; color: #FFA500; font-weight: bold;"> 3.2 || Top 10 Sales Games in North America </div>

---



In [43]:
# Filter the DataFrame for years between 2015 and 2020
filtered_df = df[(df['Year'] >= 2015) & (df['Year'] <= 2020)]

# Group by 'Year', 'Name', 'Platform', 'Genre', and 'Publisher', and sum the 'Global_Sales'
top_10_games_North_America_sales = filtered_df.groupby(['Year', 'Name', 'Platform', 'Genre', 'Publisher'])['NA_Sales'].sum().nlargest(15)
top_10_games_North_America_sales

Year  Name                                     Platform  Genre         Publisher                            
2015  Call of Duty Black Ops Three             PSFour    Shooter       Activision                               5.77
      Star Wars Battlefront TwoZeroOneFive     PSFour    Shooter       Electronic Arts                          2.93
      Halo Five Guardians                      XOne      Shooter       Microsoft Game Studios                   2.64
      NBA TwoKOneSix                           PSFour    Sports        TakeTwo Interactive                      2.48
      Fallout Four                             PSFour    Role Playing  Bethesda Softworks                       2.47
      Gears of War Ultimate Edition            XOne      Shooter       Microsoft Game Studios                   2.38
      Madden NFL OneSix                        PSFour    Sports        Electronic Arts                          2.33
      Uncharted The Nathan Drake Collection    PSFour    Action        S

|    |   Year | Name                                      | Platform   | Genre      | Publisher                             |   NA_Sales |
|----|--------|-------------------------------------------|------------|------------|---------------------------------------|------------|
|  0 |   2015 | Call of Duty Black Ops Three              | PSFour     | Shooter    | Activision                            |       5.77 |
|  1 |   2015 | Star Wars Battlefront TwoZeroOneFive      | PSFour     | Shooter    | Electronic Arts                       |       2.93 |
|  2 |   2015 | Halo Five Guardians                       | XOne       | Shooter    | Microsoft Game Studios                |       2.64 |
|  3 |   2015 | NBA TwoKOneSix                            | PSFour     | Sports     | TakeTwo Interactive                   |       2.48 |
|  4 |   2015 | Fallout Four                              | PSFour     | Role Playing | Bethesda Softworks                  |       2.47 |
|  5 |   2015 | Gears of War Ultimate Edition             | XOne       | Shooter    | Microsoft Game Studios                |       2.38 |
|  6 |   2015 | Madden NFL OneSix                         | PSFour     | Sports     | Electronic Arts                       |       2.33 |
|  7 |   2015 | Uncharted The Nathan Drake Collection     | PSFour     | Action     | Sony Computer Entertainment           |       1.96 |
|  8 |   2015 | Splatoon                                  | WiiU       | Shooter    | Nintendo                              |       1.55 |
|  9 |   2015 | Batman Arkham Knight                      | PSFour     | Action     | Warner Bros Interactive Entertainment |       1.49 |
| 10 |   2015 | Mortal Kombat X                           | PSFour     | Fighting   | Warner Bros Interactive Entertainment |       1.47 |
| 11 |   2016 | Uncharted Four A Thiefs End               | PSFour     | Shooter    | Sony Computer Entertainment           |       1.30 |
| 12 |   2016 | Tom Clancys The Division                  | PSFour     | Shooter    | Ubisoft                               |       1.28 |
| 13 |   2015 | Super Mario Maker                         | WiiU       | Platform   | Nintendo                              |       1.18 |
| 14 |   2015 | The Legend of Zelda Majoras Mask ThreeD   | ThreeDS    | Action     | Nintendo                              |       1.14 |





---
<div style="margin-top: 20px; font-family: Arial, sans-serif; font-size: 22px; color: #FFA500; font-weight: bold;"> 3.3 || Top 10 Sales Games in EU</div>

---



In [45]:
# Filter the DataFrame for years between 2015 and 2020
filtered_df = df[(df['Year'] >= 2015) & (df['Year'] <= 2020)]

# Group by 'Year', 'Name', 'Platform', 'Genre', and 'Publisher', and sum the 'Global_Sales'
top_10_games_EU_sales = filtered_df.groupby(['Year', 'Name', 'Platform', 'Genre', 'Publisher'])['EU_Sales'].sum().nlargest(15)
top_10_games_EU_sales

Year  Name                                   Platform  Genre         Publisher                            
2015  FIFA OneSix                            PSFour    Sports        Electronic Arts                          6.06
      Call of Duty Black Ops Three           PSFour    Shooter       Activision                               5.81
2016  FIFA OneSeven                          PSFour    Sports        Electronic Arts                          3.75
2015  Star Wars Battlefront TwoZeroOneFive   PSFour    Shooter       Electronic Arts                          3.29
      Fallout Four                           PSFour    Role Playing  Bethesda Softworks                       3.15
2016  Uncharted Four A Thiefs End            PSFour    Shooter       Sony Computer Entertainment              2.07
2015  The Witcher Three Wild Hunt            PSFour    Role Playing  Namco Bandai Games                       2.00
      Assassins Creed Syndicate              PSFour    Action        Ubisoft            

|    |   Year | Name                                   | Platform   | Genre         | Publisher                             |   EU_Sales |
|----|--------|----------------------------------------|------------|---------------|---------------------------------------|------------|
|  0 |   2015 | FIFA OneSix                            | PSFour     | Sports        | Electronic Arts                       |       6.06 |
|  1 |   2015 | Call of Duty Black Ops Three           | PSFour     | Shooter       | Activision                            |       5.81 |
|  2 |   2016 | FIFA OneSeven                          | PSFour     | Sports        | Electronic Arts                       |       3.75 |
|  3 |   2015 | Star Wars Battlefront TwoZeroOneFive   | PSFour     | Shooter       | Electronic Arts                       |       3.29 |
|  4 |   2015 | Fallout Four                           | PSFour     | Role Playing  | Bethesda Softworks                    |       3.15 |
|  5 |   2016 | Uncharted Four A Thiefs End            | PSFour     | Shooter       | Sony Computer Entertainment           |       2.07 |
|  6 |   2015 | The Witcher Three Wild Hunt            | PSFour     | Role Playing  | Namco Bandai Games                    |       2.00 |
|  7 |   2015 | Assassins Creed Syndicate              | PSFour     | Action        | Ubisoft                               |       1.92 |
|  8 |   2015 | Uncharted The Nathan Drake Collection | PSFour     | Action        | Sony Computer Entertainment           |       1.69 |
|  9 |   2016 | Tom Clancys The Division               | PSFour     | Shooter       | Ubisoft                               |       1.61 |
| 10 |   2015 | Batman Arkham Knight                   | PSFour     | Action        | Warner Bros Interactive Entertainment |       1.58 |
| 11 |   2015 | Need for Speed TwoZeroOneFive         | PSFour     | Racing        | Electronic Arts                       |       1.49 |
| 12 |   2015 | Metal Gear Solid V The Phantom Pain   | PSFour     | Action        | Konami Digital Entertainment          |       1.35 |
| 13 |   2015 | Halo Five Guardians                   | XOne       | Shooter       | Microsoft Game Studios                |       1.20 |
| 14 |   2015 | Tom Clancys Rainbow Six Siege         | PSFour     | Shooter       | Ubisoft                               |       1.16 |





---
<div style="margin-top: 20px; font-family: Arial, sans-serif; font-size: 22px; color: #FFA500; font-weight: bold;"> 3.4 || Top 10  Sales Games in Japan </div>

---



In [47]:
# Filter the DataFrame for years between 2015 and 2020
filtered_df = df[(df['Year'] >= 2015) & (df['Year'] <= 2020)]

# Group by 'Year', 'Name', 'Platform', 'Genre', and 'Publisher', and sum the 'Global_Sales'
top_10_games_JP_sales = filtered_df.groupby(['Year', 'Name', 'Platform', 'Genre', 'Publisher'])['JP_Sales'].sum().nlargest(15)
top_10_games_JP_sales

Year  Name                                     Platform  Genre         Publisher                   
2015  Monster Hunter X                         ThreeDS   Action        Capcom                          2.78
      Yokai Watch Busters                      ThreeDS   Action        Level Five                      2.28
      Splatoon                                 WiiU      Shooter       Nintendo                        1.44
      Animal Crossing Happy Home Designer      ThreeDS   Simulation    Nintendo                        1.39
2016  Yokai Watch Three                        ThreeDS   Action        Level Five                      1.27
2015  Super Mario Maker                        WiiU      Platform      Nintendo                        0.93
      Monster Strike ThreeDS                   ThreeDS   Action        mixi Inc                        0.86
      Rhythm Heaven The Best                   ThreeDS   Misc          Nintendo                        0.71
2016  Dragon Quest Monsters Joker Th

|    |   Year | Name                                     | Platform   | Genre        | Publisher                          |   JP_Sales |
|----|--------|------------------------------------------|------------|--------------|------------------------------------|------------|
|  0 |   2015 | Monster Hunter X                         | ThreeDS    | Action       | Capcom                             |       2.78 |
|  1 |   2015 | Yokai Watch Busters                      | ThreeDS    | Action       | Level Five                         |       2.28 |
|  2 |   2015 | Splatoon                                 | WiiU       | Shooter      | Nintendo                           |       1.44 |
|  3 |   2015 | Animal Crossing Happy Home Designer      | ThreeDS    | Simulation   | Nintendo                           |       1.39 |
|  4 |   2016 | Yokai Watch Three                        | ThreeDS    | Action       | Level Five                         |       1.27 |
|  5 |   2015 | Super Mario Maker                        | WiiU       | Platform     | Nintendo                           |       0.93 |
|  6 |   2015 | Monster Strike                           | ThreeDS    | Action       | mixi Inc                           |       0.86 |
|  7 |   2015 | Rhythm Heaven The Best                   | ThreeDS    | Misc         | Nintendo                           |       0.71 |
|  8 |   2016 | Dragon Quest Monsters Joker Three        | ThreeDS    | Role Playing | Square Enix                        |       0.63 |
|  9 |   2016 | Yokai Sangokushi                         | ThreeDS    | Action       | Level Five                         |       0.55 |
| 10 |   2015 | Fire Emblem Fates                        | ThreeDS    | Role Playing | Nintendo                           |       0.52 |
| 11 |   2015 | Metal Gear Solid V The Phantom Pain      | PSFour     | Action       | Konami Digital Entertainment       |       0.48 |
| 12 |   2015 | The Legend of Zelda Majoras Mask ThreeD  | ThreeDS    | Action       | Nintendo                           |       0.47 |
| 13 |   2016 | Kirby Planet Robobot                     | ThreeDS    | Action       | Nintendo                           |       0.44 |
| 14 |   2015 | Tales of Zestiria                        | PSThree    | Role Playing | Namco Bandai Games                 |       0.42 |





---
<div style="margin-top: 20px; font-family: Arial, sans-serif; font-size: 22px; color: #FFA500; font-weight: bold;"> 3.5 || Top 10 Sales Gamesin other Domains </div>

---



In [49]:
# Filter the DataFrame for years between 2015 and 2020
filtered_df = df[(df['Year'] >= 2015) & (df['Year'] <= 2020)]

# Group by 'Year', 'Name', 'Platform', 'Genre', and 'Publisher', and sum the 'Global_Sales'
top_10_games_other_sales = filtered_df.groupby(['Year', 'Name', 'Platform', 'Genre', 'Publisher'])['Other_Sales'].sum().nlargest(15)
top_10_games_other_sales

Year  Name                                   Platform  Genre         Publisher                            
2015  Call of Duty Black Ops Three           PSFour    Shooter       Activision                               2.31
      FIFA OneSix                            PSFour    Sports        Electronic Arts                          1.26
      Star Wars Battlefront TwoZeroOneFive   PSFour    Shooter       Electronic Arts                          1.23
      Fallout Four                           PSFour    Role Playing  Bethesda Softworks                       1.10
      Uncharted The Nathan Drake Collection  PSFour    Action        Sony Computer Entertainment              0.74
      NBA TwoKOneSix                         PSFour    Sports        TakeTwo Interactive                      0.69
2016  FIFA OneSeven                          PSFour    Sports        Electronic Arts                          0.69
      Uncharted Four A Thiefs End            PSFour    Shooter       Sony Computer Enter

|    |   Year | Name                                    | Platform   | Genre        | Publisher                             |   Other_Sales |
|----|--------|-----------------------------------------|------------|--------------|---------------------------------------|---------------|
|  0 |   2015 | Call of Duty Black Ops Three            | PSFour     | Shooter      | Activision                            |          2.31 |
|  1 |   2015 | FIFA OneSix                            | PSFour     | Sports       | Electronic Arts                       |          1.26 |
|  2 |   2015 | Star Wars Battlefront TwoZeroOneFive   | PSFour     | Shooter      | Electronic Arts                       |          1.23 |
|  3 |   2015 | Fallout Four                           | PSFour     | Role Playing | Bethesda Softworks                    |          1.1  |
|  4 |   2015 | Uncharted The Nathan Drake Collection   | PSFour     | Action       | Sony Computer Entertainment           |          0.74 |
|  5 |   2015 | NBA TwoKOneSix                         | PSFour     | Sports       | TakeTwo Interactive                   |          0.69 |
|  6 |   2016 | FIFA OneSeven                          | PSFour     | Sports       | Electronic Arts                       |          0.69 |
|  7 |   2016 | Uncharted Four A Thiefs End            | PSFour     | Shooter      | Sony Computer Entertainment           |          0.65 |
|  8 |   2015 | Batman Arkham Knight                   | PSFour     | Action       | Warner Bros Interactive Entertainment |          0.61 |
|  9 |   2015 | Madden NFL OneSix                      | PSFour     | Sports       | Electronic Arts                       |          0.59 |
| 10 |   2016 | Tom Clancys The Division               | PSFour     | Shooter      | Ubisoft                               |          0.57 |
| 11 |   2015 | The Witcher Three Wild Hunt            | PSFour     | Role Playing | Namco Bandai Games                    |          0.56 |
| 12 |   2015 | Assassins Creed Syndicate              | PSFour     | Action       | Ubisoft                               |          0.5  |
| 13 |   2015 | Metal Gear Solid V The Phantom Pain    | PSFour     | Action       | Konami Digital Entertainment          |          0.47 |
| 14 |   2015 | Mortal Kombat X                        | PSFour     | Fighting     | Warner Bros Interactive Entertainment |          0.47 |



---

![](https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRUuhiEqdW4njY8ecn_ps7nA1Ih6DH_PIhVA8kH5xouXA&s)